public/pages/DefineDetector/components/CustomResultIndex/CustomResultIndex.tsx

```diff
@@ -0,0 +1,135 @@
+        <EuiTitle size="s" id={'resultIndexField'}>
+          <h2>Custom result index</h2>
+        </EuiTitle>
+      }
+      subTitle={
+        <EuiText
+          className="content-panel-subTitle"
+          style={{ lineHeight: 'normal' }}
+        >
+          Store detector result to your own index.{' '}
+          <EuiLink href={`${BASE_DOCS_LINK}/ad`} target="_blank">
+            Learn more <EuiIcon size="s" type="popout" />
+          </EuiLink>
+        </EuiText>
+      }
+    >
+      {props.isEdit ? (
+        <EuiFlexItem>
+          <EuiCallOut
+            data-test-subj="resultIndexReadOnlyCallout"
+            title="You can't change the custom result index after you create the detector."
+            color="primary"
+            iconType="iInCircle"
+            size="s"
+          ></EuiCallOut>
+        </EuiFlexItem>
+      ) : null}
+      {props.isEdit ? <EuiSpacer size="m" /> : null}
+
+      <Field
+        name="resultIndex"
+        validate={enabled ? validateCustomResultIndex : null}
+      >
+        {({ field, form }: FieldProps) => (
+          <EuiFlexGroup direction="column">
+            <EuiFlexItem>
+              <EuiCheckbox
+                id={'resultIndexCheckbox'}
+                label="Enable custom result index"
+                checked={enabled}
+                disabled={props.isEdit}
+                onChange={() => {
+                  if (enabled) {
+                    form.setFieldValue('resultIndex', '');
+                  }
+                  setEnabled(!enabled);
+                }}
+              />
+            </EuiFlexItem>
+
+            {enabled && !props.isEdit ? (
+              <EuiFlexItem>
+                <EuiCallOut
+                  data-test-subj="cannotEditResultIndexCallout"
+                  title="You can't change the custom result index after you create the detector. You can manage the result index with Index Management plugin."
+                  color="warning"
+                  iconType="alert"
+                  size="s"
+                ></EuiCallOut>
+              </EuiFlexItem>
+            ) : null}
+
+            {enabled && !props.isEdit ? (
+              <EuiFlexItem>
+                <EuiFormRow
+                  label="Field"
+                  isInvalid={isInvalid(field.name, form)}
+                  error={getError(field.name, form)}
+                  helpText={`Custom result index name must contain less than 255 characters including the prefix "opensearch-ad-plugin-result-". Valid characters are a-z, 0-9, -(hyphen) and _(underscore).`}
+                >
+                  <EuiFieldText
+                    id="resultIndex"
+                    placeholder="Enter result index name"
+                    prepend={props.isEdit ? '' : CUSTOM_AD_RESULT_INDEX_PREFIX}
+                    disabled={props.isEdit}
+                    {...field}
+                  />
+                </EuiFormRow>
+              </EuiFlexItem>
+            ) : null}
```

Comment:
If enabled and `isEdit` is true, shouldn't the selected custom result index still show? Maybe the `!props.isEdit` condition can be removed (current conditions in `prepend` and `disabled` have some unreachable scenarios as well because of `!props.isEdit` condition).

Reply:
Good catch, change to show result index when edit detector and remove the callout of `"You can't change the custom result index after you create the detector."` as it's duplicate as the callout message `title="You can't change the custom result index after you create the detector. You can manage the result index with Index Management plugin."`. Will ask UX designer to help review.